app/models/test_run.rb

```diff
@@ -144,6 +147,9 @@ def create_test_group_results_from_json(test_output)
     json_root = nil
     begin
       json_root = JSON.parse(test_output)
+      json_root['test_groups'].each do |j|
+        j['test_group_id'] = test_groups[j['test_group_id']].name
+      end
```

Comment:
I feel like this overcomplicates things, or may potentially be wrong.
1) If the autotester passes back ids, why just not use them as-is rather than fetching names from markus' db?
2) You're fetching test_groups from db (where their id != their position in the array) with an index from the autotester, shouldn't it be find_by that 'test_group_id'?

Reply:
This is a hack because MarkUs currently requires the test_group_result's test_group attribute to be a string.  I think we should get rid of the test_group_result table completely but this is so that we can get back a result for now without having to change the db schema.